File: 1221. Split a String in Balanced Strings.java
class Solution {
    public int balancedStringSplit(String s) {
        int r = 0, l = 0, res = 0;
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i) == 'R') r++;
            else l++;

            if(r == l) res++;
        }
        return res;
    }
}

File: 1221. Split a String in Balanced Strings.ts
function balancedStringSplit(s: string): number {
    const chars = [...s];
    let balance : number = 0, res : number = 0;

    for(const char of chars){
        balance += char == 'L' ? 1 : -1;
        res += balance === 0 ? 1 : 0;
    }

    return res;
};
